update emitter readme (#3227)
Azure/autorest.csharp: src/CADL.Extension/readme.md

```diff
 "dependencies": {
+      "@azure-tools/typespec-csharp": "latest"
  },
```

Run `npm install` to install the dependency

### Generate .NET SDK

Run command `npx tsp compile --emit @azure-tools/typespec-csharp <path-to-typespec-file>`

e.g.

```cmd
npx tsp compile main.tsp --emit @azure-tools/typespec-csharp
```

or

```cmd
npx tsp compile client.tsp --emit @azure-tools/typespec-csharp
```

## Configuration

You can further configure the SDK generated, using the emitter options on @azure-tools/typespec-csharp.

You can set options in the command line directly via `--option @azure-tools/typespec-csharp.<optionName>=XXX`, e.g. `--option @azure-tools/typespec-csharp.namespace=azure.AI.DeviceUpdate`

or

Modify `tspconfig.yaml` in typespec project, add emitter options under options/@azure-tools/typespec-csharp.

```diff
emit:
  - "@azure-tools/typespec-csharp"
options:
  "@azure-tools/typespec-csharp":
+    namespace: Azure.Template.MyTypeSpecProject
```

**Supported Emitter options**:
- `namespace` define the client library namespace. e.g. Azure.IoT.DeviceUpdate.
- `emitter-output-dir` define the output dire path which will store the generated code.

> - `generate-protocol-methods` indicate if you want to generate **protocol method** for each operation. By default is true.

Reviewer: ```suggestion
- `generate-protocol-methods` indicate if you want to generate **protocol method** for every operation or not. The default option is true.
```

> - `generate-convenience-methods` indicate if you want to generate **convenience method** for each operation. By default is true.

Reviewer: Same above
